app/controllers/admin/service_contacts_controller.rb

```diff
@@ -49,13 +50,15 @@ def create
     end
 
     def destroy
-      contact = Contact.find(params[:id])
-      service = contact.service
-      location = service.location
+      service = Service.find(params[:service_id])
 
       authorize service
 
-      contact.destroy
+      resource = service.resource_contacts.find_by(contact_id: params[:id])
```

Comment:
I have largely the same aloud thoughts about slimming down this method I did as the `ContactsController`.